sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -20,21 +25,110 @@
      */
     XML;
 
+    private static final ClientLogger LOGGER = new ClientLogger(SerializerEncoding.class);
+    private static final String CONTENT_TYPE = "Content-Type";
+    private static final List<SerializerEncodingMimeTypes> SUPPORTED_MIME_TYPES;
+    private static final SerializerEncoding DEFAULT_ENCODING = JSON;
+
+    static {
+        SUPPORTED_MIME_TYPES = new ArrayList<>();
+
+        // Encodings and suffixes from: https://tools.ietf.org/html/rfc6838
+        SUPPORTED_MIME_TYPES.add(new SerializerEncodingMimeTypes(XML, "xml",
+            new MimeType("text/xml"), new MimeType("application/xml")));
+        SUPPORTED_MIME_TYPES.add(new SerializerEncodingMimeTypes(JSON, "json",
+            new MimeType("application/json")));
+    }
+
     /**
      * Determines the serializer encoding to use based on the Content-Type header.
      *
-     * @param headers the headers to check the encoding for
-     * @return the serializer encoding to use for the body
+     * @param headers the headers to check the encoding for.
+     * @return the serializer encoding to use for the body. {@link #JSON} if there is no Content-Type header or an
+     * unrecognized Content-Type encoding is returned.
      */
     public static SerializerEncoding fromHeaders(HttpHeaders headers) {
-        String mimeContentType = headers.getValue("Content-Type");
-        if (mimeContentType != null) {
-            String[] parts = mimeContentType.split(";");
-            if (parts[0].equalsIgnoreCase("application/xml") || parts[0].equalsIgnoreCase("text/xml")) {
-                return XML;
+        final String mimeContentType = headers.getValue(CONTENT_TYPE);
+        if (mimeContentType == null) {
+            LOGGER.warning("'{}' not found. Returning default encoding: {}", CONTENT_TYPE, JSON);
+            return JSON;
+        }
+
+        final String[] parts = mimeContentType.split(";");
+
+        final String mimeType = parts[0].toLowerCase(Locale.ROOT);
+        final String[] mimeTypeParts = mimeType.split("/");
+        if (mimeTypeParts.length != 2) {
+            LOGGER.warning("Content-Type '{}' does not match mime-type formatting 'type/'subtype'. "
```

Comment:
nit: This uses a slightly different error message than the same check in `MimeType`.